cdap-docs/reference-manual/source/release-notes.rst

```diff
@@ -30,6 +30,249 @@ Cask Data Application Platform Release Notes
    :backlinks: none
    :depth: 2
    
+`Release 4.3.0 <http://docs.cask.co/cdap/4.3.0/index.html>`__
+=============================================================
+
+Summary
+-------
+
+1. ** Data Pipelines:**
+	- Support for Conditional execution of parts of a Pipeline
+	- Ability for pipelines to trigger other pipelines for cross-team, cross-pipeline inter-connectivity, and to build complex interconnected pipelines.
+	- Improved Pipeline Studio with redesigned nodes, undo/redo capability, metrics
+	- Automated Upgrade of Pipelines to newer CDAP versions
+	- Custom Icons and Labels for Pipeline Plugin Developers
+	- Operational Insights into pipelines for Devops or Operations team
+
+2. ** Data Preparation:**
+	- Support for User Defined Directives (UDD), so users can write their own custom directives for cleansing/preparing data. 
+	- Restricting Directive Usage and Ability to alias Directives for your IT Administrators to control directive access
+
+3. ** Governance & Security:**
+	- Standardized authorization model 
+	- Apache Ranger Integration for IT Security Team
+
+4. ** Enhanced support for Apache Spark:**
+	- PySpark Support so data scientists can develop their Spark logic in Python, while still taking advantage of enterprise integration capabilities of CDAP
+	- Spark Dataframe Support so Spark developers can access CDAP datasets as Spark DataFrames
+
+5. ** New Frameworks and Tools:**
+	- Microservices for Real-time IoT use cases.
+	- Introducing Distributed Rules Engine - for Business Analysts for effectively manage rules for data transformation and data policy.
+
+New Features
+------------
+
+Data Pipelines Enhancements
+---------------------------
+
+- :cask-issue:`CDAP-12033` - Added a new splitter transform plugin type that can send output to different ports
+
+- :cask-issue:`CDAP-12136` - Added a union splitter transform that will send records to different ports depending on which type in the union it is.
+
+- :cask-issue:`CDAP-12137` - Added a splitter transform that splits records based on whether the specified field is null.
+
+- :cask-issue:`CDAP-12154` - Added Kafka Alert Publisher to publish alerts from a cdap pipeline to a kafka topic.
+
+- :cask-issue:`CDAP-12155` - Added a plugin that publishes alerts to CDAP TMS
+
+- :cask-issue:`CDAP-12167` - plugin prepareRun and onFinish methods now run in a separate transaction per plugin so that pipelines with many plugins will not timeout.
+
+- :cask-issue:`CDAP-12191` - All pipeline plugins now have access to the pipeline namespace and name through their context object.
+
+- :cask-issue:`CDAP-9107` - Adds feature that allows undoing and redoing of actions in pipeline Studio.
+
+- :cask-issue:`CDAP-10619` - UI: Add upgrade step when importing pipeline
```

Comment:
Added an automated UI flow for users to upgrade pipelines to newer CDAP versions